packages/typescript-estree/src/parseSettings/inferSingleRun.ts

```diff
@@ -18,13 +18,16 @@ export function inferSingleRun(options: TSESTreeOptions | undefined): boolean {
   // https://github.com/typescript-eslint/typescript-eslint/issues/9504
   // There's no support (yet?) for extraFileExtensions in single-run hosts.
   // Only watch program hosts and project service can support that.
-  if (options?.extraFileExtensions?.length) {
+  if (
+    options?.extraFileExtensions?.length &&
+    (!options.projectService || options.project)
```

Comment:
[Bug] Why would `options.projectService` be negated (`!`), but not `options.project`?

```suggestion
    (!options.projectService || options.project)
```

Reply:
According to the comments, in that case `projectService` must be truthy to infer singlerun when `extraFileExtensions` is given. I'm not sure about support for project options.
+It seems like it would be better to remove the conditions for `options.project`. What do you think?